instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/__init__.py

```diff
@@ -230,6 +238,64 @@ def _traced_execute_pipeline(func, instance, args, kwargs):
 
         return response
 
+    def _add_create_attributes(span, args):
+        _set_span_attribute(
+            span, "redis.create_index.index", _value_or_none(args, 1)
+        )
+        # According to: https://github.com/redis/redis-py/blob/master/redis/commands/search/commands.py#L155 schema is last argument for execute command
+        try:
+            schema_index = args.index("SCHEMA")
+        except ValueError:
+            return
+        schema = args[schema_index:]
+        field_attribute = ""
+        # Schema in format:
+        # [first_field_name, first_field_type, first_field_some_attribute1, first_field_some_attribute2, second_field_name, ...]
+        field_types = ["NUMERIC", "TEXT", "GEO", "TAG", "VECTOR"]
+        for index in range(len(schema)):
+            if schema[index] in field_types:
+                field_attribute += (
+                    f"Field(name: {schema[index - 1]}, type: {schema[index]});"
+                )
+        _set_span_attribute(
+            span,
+            "redis.create_index.fields",
+            field_attribute,
+        )
+
+    def _add_search_attributes(span, response, args):
+        _set_span_attribute(
+            span, "redis.search.index", _value_or_none(args, 1)
+        )
+        _set_span_attribute(
+            span, "redis.search.query", _value_or_none(args, 2)
+        )
+        # Parse response from search
+        # https://redis.io/docs/latest/commands/ft.search/
+        # Response in format:
+        # [number_of_returned_documents, index_of_first_returned_doc, first_doc(as a list), index_of_second_returned_doc, second_doc(as a list) ...]
+        # Returned documents in array format:
+        # [first_field_name, first_field_value, second_field_name, second_field_value ...]
+        number_of_returned_documents = _value_or_none(response, 0)
+        _set_span_attribute(
+            span, "redis.search.total", number_of_returned_documents
+        )
+        if "NOCONTENT" in args:
```

Comment:
do we also need to early return if not number_of_returned_documents?

Reply:
We can, then we don't need another if in line 285.